homeassistant/components/rympro/sensor.py

```diff
@@ -0,0 +1,70 @@
+"""Sensor for RymPro meters."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfVolume
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import RymProDataUpdateCoordinator
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors for device."""
+    coordinator: RymProDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        [
+            RymProSensor(coordinator, meter_id, meter["read"], config_entry.entry_id)
+            for meter_id, meter in coordinator.data.items()
+        ]
+    )
+
+
+class RymProSensor(CoordinatorEntity[RymProDataUpdateCoordinator], SensorEntity):
+    """Sensor for RymPro meters."""
+
+    _attr_has_entity_name = True
+    _attr_name = "Last Read"
+    _attr_device_class = SensorDeviceClass.WATER
+    _attr_native_unit_of_measurement = UnitOfVolume.CUBIC_METERS
+    _attr_state_class = SensorStateClass.TOTAL_INCREASING
+
+    def __init__(
+        self,
+        coordinator: RymProDataUpdateCoordinator,
+        meter_id: int,
+        last_read: int,
+        entry_id: str,
+    ) -> None:
+        """Initialize sensor."""
+        super().__init__(coordinator)
+        self._meter_id = meter_id
+        self._entity_registry: er.EntityRegistry | None = None
+        unique_id = f"{entry_id}_{meter_id}"
```

Comment:
Why is the entry ID added to the unique ID? That seems odd and unneeded.

Reply:
I don't know if it's guaranteed that meter ids are unique across accounts. I thought about using the unique id instead of the entry id, but I'm a bit reluctant to do so because it's PII (the unique id is the account number, which is the account owner's national ID number).

Reply:
You could better maybe use a combination of account number and meter ID.
Or, in case you really thin it should use the entry ID instead, use just the entry ID (it is already unique, no need to add the meter ID in that case).

Please note, this should be adjusted before the final release lands, otherwise, you'd have to deal with unique ID migrations in later releases.

../Frenck

Reply:
The meter ID is needed because there can be more than one meter per account.
For the prefix, my preference for entry ID over the account number is that the former doesn't contain PII. Maybe it's not a big concern because we do use it elsewhere, but it feels a bit off to me.

Reply:
ok 👍